src/main/kotlin/solve/scene/view/PlaneView.kt

```diff
@@ -139,17 +176,101 @@ class PlaneView(
     }
 
     private fun addListeners() {
-        canvasNode.addEventFilter(MouseEvent.MOUSE_PRESSED, mousePressedHandler)
+        canvasNode.addEventHandler(MouseEvent.MOUSE_PRESSED, mousePressedHandler)
         canvasNode.addEventHandler(MouseEvent.MOUSE_RELEASED, mouseReleasedHandler)
+
+        plane.layerSettings.enabledProperty.addListener(weakEnabledChangedEventHandler)
     }
 
     private fun removeListeners() {
         canvasNode.removeEventHandler(MouseEvent.MOUSE_PRESSED, mousePressedHandler)
         canvasNode.removeEventHandler(MouseEvent.MOUSE_RELEASED, mouseReleasedHandler)
+
+        plane.layerSettings.enabledProperty.removeListener(weakEnabledChangedEventHandler)
+    }
+
+    private fun getColorWithOpacity() = plane.layerSettings.getColorWithOpacity(plane)
+
+    private fun setPlaneElementColor(newColor: Color) {
+        planeElement.color = newColor
+        addToFrameDrawer()
     }
 
-    private fun getColorWithOpacity(): Color {
-        val rgb = plane.layerSettings.getColor(plane)
-        return Color(rgb.red, rgb.green, rgb.blue, plane.layerSettings.opacity)
+    private fun redrawPlaneWithColor(newColor: Color) {
+        setPlaneElementColor(newColor)
+        frameDrawer.redrawPoints(planeElement.points)
+    }
+
+    private fun getDisabledPlaneColor() = getColorWithOpacity().withReplacedOpacity(LayerSettings.MinOpacity)
+
+    private fun onEnabledChanged() {
+        lateinit var planeVisibilityColor: Color
+
+        if (plane.layerSettings.enabled) {
+            planeVisibilityColor = lastEnabledPlaneElementColor
+            planeUIDLabel?.isVisible = true
+        } else {
+            planeVisibilityColor = Color.TRANSPARENT
+            planeUIDLabel?.isVisible = false
+        }
+
+        redrawPlaneWithColor(planeVisibilityColor)
+    }
+
+    private fun createUIDLabel(): Label {
+        val uidLabel = Label(plane.uid.toString())
+        uidLabel.textFill = getBlackOrWhiteContrastingTo(plane.layerSettings.getColor(plane))
+        uidLabel.font = Font.font(null, FontWeight.BOLD, UIDLabelFontSize)
+
+        return uidLabel
+    }
+
+    private fun calculatePlaneCenterPoint(): DoublePoint {
```

Comment:
On min scale label doesn't fit in the frame 
![image](https://user-images.githubusercontent.com/55746901/229477646-fc03726f-b17f-4cc4-a3b4-a78c6ff68a29.png)

Reply:
Looks strange when there are a few overlapped planes.
![image](https://user-images.githubusercontent.com/55746901/229477849-ef1dc291-edde-4f60-a6c3-06cf5276f257.png)

Reply:
> Looks strange when there are a few overlapped planes. ![image](https://user-images.githubusercontent.com/55746901/229477849-ef1dc291-edde-4f60-a6c3-06cf5276f257.png)

We need to think about how we want to display a difference between uid labels of the overlapped planes. In the current implementation, the uid label is a node and plane is a canvas image, so the overlap check is a complicated task. I've created an issue

Reply:
> On min scale label doesn't fit in the frame 

Fixed:
![labels_scale](https://user-images.githubusercontent.com/66365072/230471307-19bbb64b-e00f-45ad-a35f-916f72dc3f27.gif)